torchvision/models/detection/retinanet.py

```diff
@@ -0,0 +1,572 @@
+import math
+from collections import OrderedDict
+
+import torch
+from torch import nn
+import torch.nn.functional as F
+from torch.jit.annotations import Dict, List, Tuple
+
+from ..utils import load_state_dict_from_url
+
+from . import _utils as det_utils
+from .anchor_utils import AnchorGenerator
+from .transform import GeneralizedRCNNTransform
+from .backbone_utils import resnet_fpn_backbone
+from ...ops.feature_pyramid_network import LastLevelP6P7
+from ...ops import boxes as box_ops
+
+
+__all__ = [
+    "RetinaNet", "retinanet_resnet50_fpn",
+]
+
+
+class RetinaNetHead(nn.Module):
+    """
+    A regression and classification head for use in RetinaNet.
+
+    Arguments:
+        in_channels (int): number of channels of the input feature
+        num_anchors (int): number of anchors to be predicted
+        num_classes (int): number of classes to be predicted
+    """
+
+    def __init__(self, in_channels, num_anchors, num_classes):
+        super(RetinaNetHead, self).__init__()
+        self.classification_head = RetinaNetClassificationHead(in_channels, num_anchors, num_classes)
+        self.regression_head = RetinaNetRegressionHead(in_channels, num_anchors)
+
+    def compute_loss(self, targets, head_outputs, anchors, matched_idxs):
+        return {
+            'classification': self.classification_head.compute_loss(targets, head_outputs, anchors, matched_idxs),
+            'bbox_regression': self.regression_head.compute_loss(targets, head_outputs, anchors, matched_idxs),
+        }
+
+    def forward(self, x):
+        return {
+            'cls_logits': self.classification_head(x),
+            'bbox_regression': self.regression_head(x)
+        }
+
+
+def sigmoid_focal_loss(
```

Comment:
Can you move this to `torchvision.ops`?

Reply:
Sure thing :+1: 

What about the type hinting here, should I replace it with `# type: (Tensor, Tensor, float, float, str) -> Tensor` ?